18_regexp_lab/01.py

```diff
@@ -9,8 +9,9 @@
 
 with open(kuku.path, 'r') as f:
     for line in f:
-        if kuku.key in line:
-            if re.search('^\s+#', line):
-                continue
+        temp = line.split()
+        if temp[0].startswith('#'):
+            continue
+        elif temp[0] == kuku.key:
             res = re.search("=\s+(\w*)", line)
             print(res.group(1))

```

Comment:
i have already uploaded a new (attitude) code (please check):

import argparse
import string
import re

parser = argparse.ArgumentParser(description='Returns a parameter that match a given key')
parser.add_argument('-p','--path', help='definitions file', required = True)
parser.add_argument('-k','--key', help='key', required = True)
kuku = parser.parse_args()

#Finally:
with open(kuku.path, 'r') as f:
    for line in f:
        prefix = re.match(r'\s*#', line)
        if prefix is not None:
            continue
        m = re.search(r'(\w+)\s*=\s*(.*)', line)
        if m.group(1) == kuku.key and m is not None:
            print(m.group(1), '==>', m.group(2) )